great_expectations/rule_based_profiler/data_assistant_result/data_assistant_result.py

```diff
@@ -1970,53 +2014,205 @@ def _get_expect_domain_values_to_be_between_line_chart(
                 x=batch_plot_component.plot_on_axis(),
                 y=min_value_plot_component.plot_on_axis(),
                 y2=alt.Y2(
-                    max_value_plot_component.name, title=metric_plot_component.title
+                    max_value_plot_component.name, title=max_value_plot_component.title
                 ),
             )
             .properties(title=title)
         )
 
-        line: alt.Chart = DataAssistantResult._get_line_chart(
-            df=df,
-            metric_plot_component=metric_plot_component,
+        if "quantiles" in df.columns:
+            lower_limit = lower_limit.encode(detail="quantiles")
+            upper_limit = upper_limit.encode(detail="quantiles")
+            band = band.encode(detail="quantiles")
+
+        metric_name: str
```

Comment:
I don't think if we gain anything annotating these types outside the for loop. That is, I don't think python will allocate this once instead of per iteration. I'd move the annotations inside to the assignment points like you do for `point_color_condition`.

Reply:
This is another case where I'm doing it [this way](https://peps.python.org/pep-0526/#global-and-local-variable-annotations). If that is what we decide to do ten the `point_color_condition` should be outside the loop as well.